redfish-core/lib/power_subsystem.hpp

```diff
@@ -25,6 +25,47 @@ namespace redfish
 inline void getPowerSubsystemAllocation(
     const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
 {
+    // Get max from CapLimits
+    sdbusplus::asio::getAllProperties(
+        *crow::connections::systemBus, "org.open_power.OCC.Control",
+        "/xyz/openbmc_project/control/host0/power_cap_limits",
+        "xyz.openbmc_project.Control.Power.CapLimits",
+        [asyncResp](const boost::system::error_code& ec,
+                    const dbus::utility::DBusPropertiesMap& propertiesList) {
+            if (ec)
+            {
+                if (ec.value() != EBADR)
+                {
+                    BMCWEB_LOG_ERROR("DBUS response error: {}", ec.value());
+                    messages::internalError(asyncResp->res);
+                }
+                return;
+            }
+
+            const uint32_t* minCap = nullptr;
+            const uint32_t* maxPowerCapValue = nullptr;
+            const bool success = sdbusplus::unpackPropertiesNoThrow(
+                dbus_utils::UnpackErrorPrinter(), propertiesList,
+                "MinPowerCapValue", minCap, "MaxPowerCapValue",
+                maxPowerCapValue);
+            if (!success)
+            {
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            // If MaxPowerCapValue valid, store Allocation properties in JSON
+            if ((maxPowerCapValue != nullptr) &&
+                (*maxPowerCapValue < UINT32_MAX))
```

Comment:
I think this is better to use numeric_limits like
```
(*maxPowerCapValue < std::numeric_limits<uint32_t>::max()))
```

Reply:
Since this is already defined as uint32, check is not really needed. Done.